src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.Log.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.WebSockets;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Sockets.Internal.Transports
+{
+    public partial class WebSocketsTransport
+    {
+        private static class Log
+        {
+            private static readonly Action<ILogger, Exception> _socketOpened =
+                LoggerMessage.Define(LogLevel.Information, new EventId(1, "SocketOpened"), "Socket opened.");
+
+            private static readonly Action<ILogger, Exception> _socketClosed =
+                LoggerMessage.Define(LogLevel.Information, new EventId(2, "SocketClosed"), "Socket closed.");
+
+            private static readonly Action<ILogger, WebSocketCloseStatus?, string, Exception> _clientClosed =
+                LoggerMessage.Define<WebSocketCloseStatus?, string>(LogLevel.Debug, new EventId(3, "ClientClosed"), "Client closed connection with status code '{status}' ({description}). Signaling end-of-input to application.");
+
+            private static readonly Action<ILogger, Exception> _waitingForSend =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(4, "WaitingForSend"), "Waiting for the application to finish sending data.");
+
+            private static readonly Action<ILogger, Exception> _failedSending =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(5, "FailedSending"), "Application failed during sending. Sending InternalServerError close frame.");
+
+            private static readonly Action<ILogger, Exception> _finishedSending =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(6, "FinishedSending"), "Application finished sending. Sending close frame.");
+
+            private static readonly Action<ILogger, Exception> _waitingForClose =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(7, "WaitingForClose"), "Waiting for the client to close the socket.");
+
+            private static readonly Action<ILogger, Exception> _closeTimedOut =
+                LoggerMessage.Define(LogLevel.Debug, new EventId(8, "CloseTimedOut"), "Timed out waiting for client to send the close frame, aborting the connection.");
+
+            private static readonly Action<ILogger, WebSocketMessageType, int, bool, Exception> _messageReceived =
+                LoggerMessage.Define<WebSocketMessageType, int, bool>(LogLevel.Debug, new EventId(9, "MessageReceived"), "Message received. Type: {messageType}, size: {size}, EndOfMessage: {endOfMessage}.");
+
+            private static readonly Action<ILogger, int, Exception> _messageToApplication =
+                LoggerMessage.Define<int>(LogLevel.Debug, new EventId(10, "MessageToApplication"), "Passing message to application. Payload size: {size}.");
```

Comment:
How are you determining what is debug vs trace? This should be trace no?

Reply:
Debug - Usable by a human who wants a lot of spam
Trace - Only really usable by a machine.

Yes, this should be Trace. I didn't do a whole review, but will now, since my build is failing to load `libuv` so hey, I've got time...